apps/nextjs/src/components/AppComponents/AdditionalMaterials/StepLayouts/StepThree.tsx

```diff
@@ -32,16 +35,35 @@ import { Glossary } from "../../AdditionalMaterials/Glossary";
 import InlineButton from "../InlineButton";
 import ResourcesFooter from "../ResourcesFooter";
 
+const log = aiLogger("additional-materials");
+
 const StepThree = () => {
   const generation = useResourcesStore(generationSelector);
+
   const docType = useResourcesStore(docTypeSelector);
   const isResourcesLoading = useResourcesStore(isResourcesLoadingSelector);
   const { setStepNumber, refineMaterial } = useResourcesActions();
   const [isFooterAdaptOpen, setIsFooterAdaptOpen] = useState(false);
+  const { downloadMaterial, setIsResourceDownloading } = useResourcesActions();
+  const isDownloading = useResourcesStore(isResourcesDownloadingSelector);
 
   const fetchMaterial =
     trpc.additionalMaterials.generateAdditionalMaterial.useMutation();
 
+  const handleDownload = async () => {
```

Comment:
Re. naming i wonder if naming is confusing here with the other `handleDownload` function introduced inthis PR

Could this be `safelyDownloadMaterial` (in the sense that it's wrapped in a try catch and checks generation/docType first?)

Or something like that?

Or
- attemptDownloadMaterial
- tryDownloadMaterial
- handleDownloadMaterial

What do you think?

Reply:
Or `onDownloadZipClick`

Reply:
`downloadClickHandler`